lib/data/datasource_impl/contact/phonebook_contact_datasource_impl.dart

```diff
@@ -25,5 +27,48 @@ class PhonebookContactDatasourceImpl implements PhonebookContactDatasource {
           ],
         )
         .toList();
+
+    return _removeDuplicatedPhoneNumbers(listAllContacts);
+  }
+
+  List<Contact> _removeDuplicatedPhoneNumbers(List<Contact> listContacts) {
+    final listVisitedPhoneNumbers = <String>[];
+    final listFilteredContacts = <Contact>[];
+
+    for (final contact in listContacts) {
+      final phoneNumber = contact.phoneNumber;
+      if (phoneNumber == null) continue;
```

Comment:
IMO, you can do `listContactsHasPhoneNumber = listContacts.where(contact.phoneNumber != null)`, outside of for loop, then remove the condition